tests/test_bench.py

```diff
@@ -797,3 +842,109 @@ def test_assert_never_runtime() -> None:
+
+        if level_enum == BenchmarkLevel.MAPPED and target_obj:
+            assert qc_mirror.num_qubits == qc_base.num_qubits
+        else:
+            assert qc_mirror.num_qubits == logical_circuit_size
+
+        assert qc_mirror.num_clbits == qc_base.num_clbits
+
+        if qc_base.num_clbits > 0:
+            assert any(inst.operation.name == "measure" for inst in qc_mirror.data)
+        else:
+            assert not any(inst.operation.name == "measure" for inst in qc_mirror.data)
+
+        assert any(inst.operation.name == "barrier" for inst in qc_mirror.data), (
+            f"Mirror circuit for level '{level_enum.name}' should contain a barrier."
+        )
+
+        # --- Layout Verification---
+        if level_enum == BenchmarkLevel.MAPPED and target_obj:
+            assert isinstance(qc_base.layout, TranspileLayout), (
+                f"Base mapped circuit for {benchmark_name} lacks TranspileLayout."
+            )
+            assert isinstance(qc_mirror.layout, TranspileLayout), (
+                f"Mirror of mapped circuit for {benchmark_name} lacks TranspileLayout."
+            )
+
+            assert qc_mirror.layout.initial_layout == qc_base.layout.initial_layout, (
+                f"Mirror circuit's initial_layout component ({qc_mirror.layout.initial_layout}) "
+                f"differs from base circuit's initial_layout ({qc_base.layout.initial_layout})."
+            )
+
+            assert qc_mirror.layout.final_layout == qc_base.layout.initial_layout, (
+                f"Mirror circuit's final_layout ({qc_mirror.layout.final_layout}) "
+                f"should revert to the base circuit's initial_layout ({qc_base.layout.initial_layout})."
+            )
+        elif qc_base.layout is None:
+            assert qc_mirror.layout is None, (
+                f"Mirror circuit for {level_enum.name} should have None layout if base was None."
+            )
+
+        # --- Verification of U @ U_inv being Identity ---
+        qc_mirror.remove_final_measurements(inplace=True)
+
+        qc_mirror.data = [instr for instr in qc_mirror.data if instr.operation.name != "barrier"]
+
+        optimized_circuit = transpile(
+            qc_mirror,
+            optimization_level=2,  # optimization_level=2 required for U@U_inv identity check; level 1 insufficient.
+            basis_gates=["u", "cx", "id"],
+        )
+
+        all_ops_are_identity_or_empty = all(instr.operation.name == "id" for instr in optimized_circuit.data)
+        assert all_ops_are_identity_or_empty, (
+            f"Unitary part of mirror (U@U_inv, MQT Bench barrier removed for this check) "
+            f"for level '{level_enum.name}' ({qc_mirror.num_qubits} qubits) "
+            f"did not optimize to only identity gates or an empty circuit. "
+            f"Found operations: {[op.operation.name for op in optimized_circuit.data if op.operation.name != 'id']}. "
+            f"Optimized QASM: {dumps(optimized_circuit)}"
+        )
```

Comment:
Just out of interest:
Would removing the Id hate from the gateset completely reduce the circuit to an empty circuit?
I'd very much prefer that.